proposals/memory-specializations/utilities.tex

```diff
@@ -0,0 +1,363 @@
+%!TEX root = std.tex
+\setcounter{chapter}{19}
+\rSec0[utilities]{General utilities library}
+
+\setcounter{section}{6}
+\rSec1[memory]{Memory}
+\synopsis{Header \tcode{<\added{experimental/ranges/}memory>} synopsis}
+
+\setcounter{subsection}{11}
+\rSec2[specialized.algorithms]{Specialized algorithms}
+
+\rSec3[uninitialized.construct.default]{uninitialized_default_construct}
+\begin{codeblock}
+namespace std { @\added{namespace experimental \{ namespace ranges \{ inline namespace v\replacedtxt{1}{2} \{}@
+\end{codeblock}
+
+{\color{newclr}
+\begin{codeblock}
+template <class I, class T>
+concept bool __ReferenceTo() {
+  return InputIterator<I>() &&
+    is_lvalue_reference_v<reference_t<I>> &&
+    Same<remove_cv_t<remove_reference_t<reference_t<I>>>, T>();
+}
+\end{codeblock}
+} %% \color{newclr}
+
+\begin{codeblock}
+  template <@\changed{class ForwardIterator}{OutputIterator O, Sentinel<O> S}@>
+    @\added{requires DefaultConstructible<value_type_t<O>{>}() \&\&}@
+             @\replacedtxt{Same<value_type_t<O>\&, reference_t<O>{>}}{__ReferenceTo<O, value_type_t<O>{>}()}@
+    @\changed{void}{O}@ uninitialized_default_construct(@\changed{ForwardIterator}{O}@ first, @\changed{ForwardIterator}{S}@ last);
```

Comment:
AFAICT this algorithm doesn't require `Writable` since it is constructing objects in-place. Suggest constraining this with `template<InputIterator O, Sentinel<O> S>`. Also, given the definition of `__ReferenceTo`, this algorithm accepts ranges whose `reference_t` is a `T const &`. That seems a bit odd to me. Then again, even `T&` is a lie if the memory is actually uninitialized.